### ISSUE TITLE
Empty frame mode() causes interpreter segmentation fault and core dump

### ISSUE THREAD

**Issue by dvorka** [h2oai/datatable#1995]
- Did you find a bug in datatable, or maybe the bug found you?
`mode()` invocation on  empty frame causes interpreter segmentation fault and core dump (while for instance `mean()` works as expected).

- How to reproduce the bug?

```
(env) dvorka@dvorka:~/h/dai/git/h2oai$ python
Python 3.6.8 (default, Jan 14 2019, 11:02:34) 
[GCC 8.0.1 20180414 (experimental) [trunk revision 259383]] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import datatable
>>> dtf = datatable.Frame({'a':[1,2,3,4], 'b':[10,20,30,40], 'c':[100,200,300,400]})
>>> dtc = dtf[datatable.f['a'] > 10,:]
>>> dtc.mode()
Segmentation fault (core dumped)
```

- What was the expected behavior?

```
(env) dvorka@dvorka:~/h/dai/git/h2oai$ python
Python 3.6.8 (default, Jan 14 2019, 11:02:34) 
[GCC 8.0.1 20180414 (experimental) [trunk revision 259383]] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import datatable
>>> dtf = datatable.Frame({'a':[1,2,3,4], 'b':[10,20,30,40], 'c':[100,200,300,400]})
>>> dtc = dtf[datatable.f['a'] > 10,:]
>>> dtc.mode()
     a   b   c
--  --  --  --
 0            

[1 row x 3 columns]
>>> 
```

- Your environment?
```
(env) dvorka@dvorka:~/h/dai/git/h2oai$ pip freeze | grep datatable
39:datatable==0.9.0.dev119
(env) dvorka@dvorka:~/h/dai/git/h2oai$ python --version
Python 3.6.8
(env) dvorka@dvorka:~/h/dai/git/h2oai$ uname -a
Linux dvorka 4.15.0-62-generic #69-Ubuntu SMP Wed Sep 4 20:55:53 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
(env) dvorka@dvorka:~/h/dai/git/h2oai$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 18.04.3 LTS
Release:	18.04
Codename:	bionic
```

- Tag the issue with `[bug]` or `[segfault]` (depending on whether it crashes
  Python or not).
:heavy_check_mark: 

- Thank you for contributing, and sorry for the inconvenience.
:wink:

**Comment by st-pasha** [comment]
Thanks for the report Martin,

It appears that this issue has been fixed a month ago in #1952. I'll add your example to the suit of unit tests though.

**Comment by dvorka** [comment]
:+1: will you close the issue?
